llvm/lib/SYCL/SYCLUtils.cpp

```diff
@@ -66,4 +69,99 @@ bool isKernelFunc(const Function *F) {
          F->hasFnAttribute("fpga.top.func");
 }
 
+void annotateKernelFunc(Function *F) {
+  F->addFnAttr("fpga.top.func", F->getName());
+  F->addFnAttr("fpga.demangled.name", F->getName());
+  F->setCallingConv(CallingConv::C);
+  F->setLinkage(llvm::GlobalValue::ExternalLinkage);
+}
+
+void removeKernelFuncAnnotation(Function *F) {
+  F->removeFnAttr("fpga.top.func");
+  F->removeFnAttr("fpga.demangled.name");
+  F->setCallingConv(CallingConv::C);
+  F->setLinkage(llvm::GlobalValue::PrivateLinkage);
+}
+
+/// Pipe are represented with 3 sting attributes here are the names:
+constexpr const char *xilinx_pipe_type =
+    "sycl_xilinx_pipe_type"; // value is "read" or "write"
+constexpr const char *xilinx_pipe_id =
+    "sycl_xilinx_pipe_id"; // value is the unique ID of the pipe
+constexpr const char *xilinx_pipe_depth =
+    "sycl_xilinx_pipe_depth"; // value is the depth of the pipe
+
+/// getAttributeAtIndex(0, ...) is the attribute on the return the first argment
+/// starts at 1
```

Comment:
```suggestion
/// getAttributeAtIndex(0, ...) is the attribute on the return. The first argument
/// starts at 1
```
What about a *working* spell-checker for 2023? :-)
I am unsure my proposition is what you mean.